lib/topological_inventory/ansible_tower/receptor/collector.rb

```diff
@@ -53,12 +52,18 @@ def async_sweep_inventory(refresh_state_uuid, sweep_scope, total_parts, refresh_
         logger.sweeping(:finish, source, sweep_scope, refresh_state_uuid)
       end
 
+      def response_received!
+        self.last_response_at = Time.now.utc
```

Comment:
Although it's not thread safe assignment, in the worst case the value can be different by miliseconds. So it doesn't matter